### PR TITLE
ITKDev: added fees list module

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

https://platform.dandigbib.org/issues/5391

#### Description

Added fees module 

#### Screenshot of the result

image

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

n/a

### CODE REVIEW

**Review by cableman — APPROVED** [review]

File: web/modules/custom/dpl_fees/src/Controller/DplFeesController.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  public function list(): array {
+    $config = [];
+
+    /** @var \Drupal\dpl_fees\Plugin\Block\FeesListBlock $plugin_block */
+    $plugin_block = $this->blockManager->createInstance('dpl_fees_list_block', $config);
+
+    $access_result = $plugin_block->access($this->currentUser());
+    if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+      throw new AccessDeniedHttpException();
+    }
```
Comment (spaceo) [inline web/modules/custom/dpl_fees/src/Controller/DplFeesController.php:67]: Todo or common way of handling access...

File: web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+   */
+  public function getFormId(): string {
+    return 'fee_list_settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $config = $this->configService->getConfig();
```
Comment (spaceo) [inline web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php:61]: Please use  `loadConfig` and change keys

File: web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      '#title' => $this->t('Payment overview url'),
+      '#default_value' => $config['paymentOverviewUrl'] ?? '',
+    ];
+
+    $form['settings']['fee_list_body_text'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Intro text'),
+      '#description' => $this->t('Display an intro-text below the headline'),
+      '#default_value' => $config['feeListBodyText'] ?? $this->t('Fees and replacement costs are handled through the new system "Mit betalingsoverblik.'),
+    ];
```
Comment (spaceo) [inline web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php:130]: Just curious: No fields are required?

**Review by spaceo — CHANGES_REQUESTED** [review]
Looks perfect! But the issue about `loadConfig` for internal use and `getConfig` for rendering is present here as well.

File: web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php (hunk 3 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php:130]: No, I could make them, but they have not been very specific about this in the specification of the system. So we'll talk to them about this during a test meeting.

File: web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php (hunk 2 above)
Comment (cableman) [reply web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php:61]: Added

File: web/modules/custom/dpl_fees/src/Controller/DplFeesController.php (hunk 1 above)
Comment (cableman) [reply web/modules/custom/dpl_fees/src/Controller/DplFeesController.php:67]: Added

File: web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php (hunk 2 above)
Comment (cableman) [reply web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php:61]: Fixed

**Pull request merged**